File: Go/1207. Unique Number of Occurrences.go
package main

//goland:noinspection GoUnusedFunction
func uniqueOccurrences(arr []int) bool {
	occurrences := make(map[int]int)

	for _, num := range arr {
		occurrences[num]++
	}

	unique := make(map[int]struct{})

	for _, count := range occurrences {
		unique[count] = struct{}{}
	}

	return len(unique) == len(occurrences)
}

File: JavaScript/1207. Unique Number of Occurrences.js
// noinspection JSUnusedGlobalSymbols

/**
 * @param {number[]} arr
 * @return {boolean}
 */
const uniqueOccurrences = (arr) => {
  const map = new Map()

  for (let i = 0; i < arr.length; i++) {
    if (map.has(arr[i])) {
      map.set(arr[i], map.get(arr[i]) + 1)
      continue
    }

    map.set(arr[i], 1)
  }

  const set = new Set(map.values())
  return set.size === map.size
}
